clang/test/CIR/CodeGen/vector-ext.cpp

```diff
@@ -1095,65 +1095,83 @@ void foo17() {
 
 void foo18() {
   vi4 a = {1, 2, 3, 4};
-  vi4 shl = a << 3;
+  int sv = 3;
```

Comment:
I guess you had to make this change to keep the `vec.splat` from being folded away. I'd like to see both forms tested.